File: ObjectProgrammingPython/Obsluga/generatorOrganizmow.py
import Zwierzeta
import Rosliny


class generatorOrganizmow:

    def klonowanie(self, organizm, swiat):
        if type(organizm) is Zwierzeta.wilk.Wilk:
            return Zwierzeta.wilk.Wilk(swiat, -1, -1, True)
        if type(organizm) is Zwierzeta.owca.Owca:
            return Zwierzeta.owca.Owca(swiat, -1, -1, True)
        if type(organizm) is Zwierzeta.antylopa.Antylopa:
            return Zwierzeta.antylopa.Antylopa(swiat, -1, -1, True)
        if type(organizm) is Zwierzeta.lis.Lis:
            return Zwierzeta.lis.Lis(swiat, -1, -1, True)
        if type(organizm) is Zwierzeta.zolw.Zolw:
            return Zwierzeta.zolw.Zolw(swiat, -1, -1, True)
        if type(organizm) is Rosliny.trawa.Trawa:
            return Rosliny.trawa.Trawa(swiat, -1, -1, True)
        if type(organizm) is Rosliny.mlecz.Mlecz:
            return Rosliny.mlecz.Mlecz(swiat, -1, -1, True)
        if type(organizm) is Rosliny.guarana.Guarana:
            return Rosliny.guarana.Guarana(swiat, -1, -1, True)
        if type(organizm) is Rosliny.jagody.Jagody:
            return Rosliny.jagody.Jagody(swiat, -1, -1, True)
        if type(organizm) is Rosliny.barszcz.Barszcz:
            return Rosliny.barszcz.Barszcz(swiat, -1, -1, True)
        if type(organizm) is Zwierzeta.cyberOwca.CyberOwca:
            return Zwierzeta.cyberOwca.CyberOwca(swiat, -1, -1, True)

File: ObjectProgrammingJava/projekt po - 2/src/projekt/obsluga/GeneratorOrganizmow.java
package projekt.obsluga;

import projekt.Organizm;
import projekt.rosliny.*;
import projekt.swiat.Swiat;
import projekt.zwierzeta.*;

import java.util.zip.ZipError;

public class GeneratorOrganizmow {

    public Organizm klonowanie(Organizm org, Swiat swiat){
        if(org instanceof Wilk) {
            return new Wilk(swiat, -1, -1, true);
        }
        if(org instanceof Owca) {
            return new Owca(swiat, -1, -1, true);
        }
        if(org instanceof Lis) {
            return new Lis(swiat, -1, -1, true);
        }
        if(org instanceof Antylopa) {
            return new Antylopa(swiat, -1, -1, true);
        }
        if(org instanceof Zolw) {
            return new Zolw(swiat, -1, -1, true);
        }
        if(org instanceof Trawa) {
            return new Trawa(swiat, -1, -1, true);
        }
        if(org instanceof Mlecz) {
            return new Mlecz(swiat, -1, -1, true);
        }
        if(org instanceof Jagody) {
            return new Jagody(swiat, -1, -1, true);
        }
        if(org instanceof Barszcz) {
            return new Barszcz(swiat, -1, -1, true);
        }
        if(org instanceof Guarana) {
            return new Guarana(swiat, -1, -1, true);
        }
        return null;
    }
}
